pyfolio/timeseries.py

```diff
@@ -752,6 +752,131 @@ def perf_stats(returns, factor_returns=None):
+        Daily noncumulative returns of the benchmark.
+         - This is in the same style as returns.
+        If None, do not compute alpha, beta, and information ratio.
+    return_stats : boolean (optional)
+        If True, returns a DataFrame of mean, 5 and 95 percentiles for
+        each perf metric.
+        If False, returns a DataFrame with the bootstrap samples for
+        each perf metric.
+
+    Returns
+    -------
+    pd.DataFrame
+        if return_stats is True:
+        - Distributional statistics of bootstrapped sampling
+        distribution of performance metrics.
+        if return_stats is False:
+        - Bootstrap samples for each performance metric.
+    """
+    bootstrap_values = OrderedDict()
+
+    for stat_func in SIMPLE_STAT_FUNCS:
+        stat_name = stat_func.__name__
+        bootstrap_values[stat_name] = calc_bootstrap(stat_func,
+                                                     returns)
+
+    if factor_returns is not None:
+        for stat_func in FACTOR_STAT_FUNCS:
+            stat_name = stat_func.__name__
+            bootstrap_values[stat_name] = calc_bootstrap(stat_func,
+                                                         returns,
+                                                         factor_returns)
+    bootstrap_values = pd.DataFrame(bootstrap_values)
+
+    if return_stats:
+        stats = bootstrap_values.apply(calc_distribution_stats)
+        return stats.T[['mean', '5%', '95%']]
+    else:
+        return bootstrap_values
+
+
+def calc_bootstrap(func, returns, *args, **kwargs):
+    """Performs a bootstrap analysis on a user-defined function returning
+    a summary statistic.
+
+    Parameters
+    ----------
+    func : function
+        Function that either takes a single array (commonly returns)
+        or two arrays (commonly returns and factor returns) and
+        returns a single value (commonly a summary
+        statistic). Additional args and kwargs are passed as well.
+    returns : pd.Series
+        Daily returns of the strategy, noncumulative.
+         - See full explanation in tears.create_full_tear_sheet.
+    factor_returns : pd.Series (optional)
+        Daily noncumulative returns of the benchmark.
+         - This is in the same style as returns.
+    n_samples : int (optional)
+        Number of bootstrap samples to draw. Default is 1000.
+        Increasing this will lead to more stable / accurate estimates.
+    Returns
+    -------
+    numpy.ndarray
+        Bootstrapped sampling distribution of passed in func.
+    """
+
+    n_samples = kwargs.pop('n_samples', 1000)
+    out = np.empty(n_samples)
+
+    if len(args) > 0:
+        factor_returns = args[0]
+        args = args[1:]
+        # Ensure that factor_returns have same index
+        factor_returns = factor_returns[returns.index]
+    else:
+        factor_returns = None
+
+    for i in range(n_samples):
+        idx = np.random.randint(len(returns), size=len(returns))
```

Comment:
Neither am I but my hunch is that the difference won't be huge. There are various different ways to do a blocked bootstrap and we might want to add these. I'd lean towards sticking with the simplest one for now and if the need arises we can add different sampling schemes later.